src/client/components/pages/Courses/MeetingTimesList.tsx

```diff
@@ -0,0 +1,472 @@
+import { MeetingRoomResponse } from 'common/dto/meeting/MeetingResponse.dto';
+import {
+  convert12To24HourTime,
+  convertTo12HourDisplayTime,
+} from 'common/utils/timeHelperFunctions';
+import {
+  BorderlessButton,
+  Button,
+  ButtonDropdownMenu,
+  ButtonDropdownMenuItem,
+  Dropdown,
+  fromTheme,
+  TextInput,
+  ValidationErrorMessage,
+  VARIANT,
+} from 'mark-one';
+
+interface MeetingTimesListProps {
+  /**
+   * The current existing meetings for this course instance
+   */
+  meetings: CourseInstanceResponseMeeting[];
+  saving: boolean;
+}
+
+interface StyledMeetingRowProps {
+  /** If true, the user is currently editing a meeting entry */
+  isRowExpanded: boolean;
+}
+
+const generateGrid = (isRowExpanded: boolean): string => {
+  if (isRowExpanded) {
+    return `"delete day timeslot start end ."
+            ". error error error error error"
+            ". room room room room room"
+            ". . . . showclose showclose"`;
+  }
+  return '"delete info edit"';
+};
+
+const StyledMeetingRow = styled.li<StyledMeetingRowProps>`
+  display: grid;
+  align-items: center;
+  border-bottom: ${fromTheme('border', 'light')};
+  grid-template-areas: ${({ isRowExpanded }) => (
+    generateGrid(isRowExpanded)
+  )};
+  grid-template-columns: min-content;
+  grid-template-rows: ${({ isRowExpanded }) => (
+    isRowExpanded
+      ? 'repeat(4, minmax(22px, auto));'
+      : ''
+  )};
+  padding: ${fromTheme('ws', 'xsmall')} 0px;
+`;
+
+const StyledMeetingInfo = styled.span`
+  grid-area: info;
+`;
+
+const StyledDay = styled.span`
+  grid-area: day;
+  display: inline-block;
+  max-width: max-content;
+  min-width: max-content;
+`;
+
+const StyledTimeslot = styled.span`
+  grid-area: timeslot;
+  height: 100%;
+  justify-self: end;
+`;
+
+const StyledStart = styled.span`
+  grid-area: start;
+  min-width: max-content;
+`;
+
+const StyledEnd = styled.span`
+  grid-area: end;
+  min-width: max-content;
+`;
+
+const StyledError = styled.span`
+    grid-area: error;
+    justify-self: start;
+`;
+
+const StyledRoom = styled.div`
+    grid-area: room;
+`;
+
+const StyledDeleteButton = styled.span`
+  grid-area: delete;
+  justify-self: start;
+`;
+
+const StyledEditButton = styled.span`
+  grid-area: edit;
+  justify-self: end;
+`;
+
+const StyledShowCloseButtons = styled.div`
+    grid-area: showclose;
+    justify-self: end;
+    > * {
+      margin-left: ${fromTheme('ws', 'xsmall')};
+    }
+`;
+
+/**
+ * The meeting times section of the modal that allows users to edit meetings
+ * for a particular course instance.
+ */
+export const MeetingTimesList
+: FunctionComponent<MeetingTimesListProps> = function ({
+  meetings,
+  saving,
+}): ReactElement {
+  /**
+   * Keeps track of the current meetings for this instance. This is updated as
+   * users add and edit meetings in the modal.
+   */
+  const [
+    currentMeetings,
+    setCurrentMeetings,
+  ] = useState(meetings);
```

Comment:
I made an update so that the state of the current meetings lives in the `MeetingModal` instead of in `MeetingTimesList`. There is now an `onChange` prop that gets called to update the meetings state. The `getCurrentMeetings` function is used to update the state of the meetings. I think this change will help us make it easier to update the meetings state. (commit de3f7f8)